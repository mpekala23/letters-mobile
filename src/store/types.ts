import { UserState } from "./User/UserTypes";
import { NotifState } from "./Notif/NotifTypes";
import { ContactState } from "./Contact/ContactTypes";

export interface AppState {
  user: UserState;
  notif: NotifState;
  contact: ContactState;
}
