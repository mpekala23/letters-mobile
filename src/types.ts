// TODO: make this typing better
// probably want a specific type for state abbrevs
export interface Facility {
  name: string;
  type: PrisonTypes;
  address: string;
  city: string;
  state: string;
  postal: string;
  phone: string;
}

export enum ProfilePicTypes {
  Topbar = 'Topbar',
  Contact = 'Contact',
  SingleContact = 'SingleContact',
}

export enum Storage {
  RememberToken = 'Ameelio-Token',
}

export enum PrisonTypes {
  State = 'State',
  Federal = 'Federal',
  Jail = 'County',
}

export enum SupportFAQTypes {
  DeleteLetter = 'DeleteLetter',
  NotArrived = 'NotArrived',
  WrongReturnAddress = 'WrongReturnAddress',
  WrongMailingAddress = 'WrongMailingAddress',
  TrackingNumber = 'TrackingNumber',
  TrackingError = 'TrackingError',
  TalkToAmeelio = 'TalkToAmeelio',
}

export enum LetterStatus {
  Draft = 'Draft',
  Created = 'Created',
  Mailed = 'Mailed',
  InTransit = 'In Transit',
  InLocalArea = 'In Local Area',
  OutForDelivery = 'Out for Delivery',
  Delivered = 'Delivered',
}

export enum LetterTypes {
  Postcard = 'postcard',
  Letter = 'letter',
}

export enum DeliveryReportTypes {
  received = 'received',
  unsure = 'unsure',
  notYetReceived = 'notYetReceived',
  haveNotAsked = 'haveNotAsked',
  haveNotReceived = 'haveNotReceived',
}

export interface Letter {
  type: LetterTypes;
  status: LetterStatus;
  isDraft: boolean;
  recipientId: number;
  content: string;
  photo?: Photo;
  letterId?: number; // TODO: Once we have more info on this field and lob, use this more
  expectedDeliveryDate?: Date;
  dateCreated?: Date;
  trackingEvents?: LetterTrackingEvent[];
}

export interface LetterTrackingEvent {
  id: number;
  name: string;
  location: ZipcodeInfo;
  date: Date;
}

export type TopbarRouteAction = {
  enabled: boolean;
  text: string;
  action: () => void | Promise<void>;
  blocking?: boolean;
};

export interface ZipcodeInfo {
  zip: string;
  city: string;
  state: string;
  lat?: number;
  long?: number;
}

export interface Photo {
  type?: 'image' | string;
  width?: number;
  height?: number;
  uri: string;
}
