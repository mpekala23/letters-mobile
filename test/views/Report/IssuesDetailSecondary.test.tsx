import React from 'react';
import { IssuesDetailSecondaryScreen } from '@views';
import { DeliveryReportTypes } from 'types';
import { render, fireEvent, toJSON } from '@testing-library/react-native';

const setup = (issueOverrides: Record<string, unknown>) => {
  const navigation = { navigate: jest.fn() };
  const route = {
    params: { issue: '', ...issueOverrides },
  };
  return {
    navigation,
    route,
    ...render(
      <IssuesDetailSecondaryScreen navigation={navigation} route={route} />
    ),
  };
};

describe('Issues Detail Secondary screen', () => {
  it('should match snapshot', () => {
    const { container } = setup(<IssuesDetailSecondaryScreen />);
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it('should navigate to Home screen when issue haveNotAsked button is pressed', async () => {
    const { navigation, getByText } = setup({
      issue: DeliveryReportTypes.haveNotAsked,
    });
    fireEvent.press(getByText('Return home'));
    expect(navigation.navigate).toHaveBeenCalledWith('ContactSelector');
  });

  // TO-DO: Test 'Call facility' button action

  it('should navigate to Home screen when issue haveNotReceived button #2 is pressed', async () => {
    const { navigation, getByText } = setup({
      issue: DeliveryReportTypes.haveNotReceived,
    });
    fireEvent.press(getByText("I'll wait"));
    expect(navigation.navigate).toHaveBeenCalledWith('ContactSelector');
  });
});
