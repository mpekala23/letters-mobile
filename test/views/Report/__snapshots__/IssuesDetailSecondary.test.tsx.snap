// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issues Detail Secondary screen should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "paddingHorizontal": 16,
          "paddingTop": 40,
        },
        Object {
          "backgroundColor": undefined,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Poppins-Bold",
          },
          Object {
            "fontSize": 20,
            "textAlign": "center",
            "width": "80%",
          },
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Poppins-Regular",
            "fontSize": 16,
          },
          Object {
            "padding": 16,
            "textAlign": "center",
          },
        ]
      }
    >
      
    </Text>
    <View
      style={
        Object {
          "marginVertical": 32,
        }
      }
    >
      <View />
    </View>
    <View
      style={
        Object {
          "width": "100%",
        }
      }
      testID="callToActionButton"
    />
  </View>
</View>
`;
